File: app/src/main/java/se/gustavkarlsson/aurora_notifier/android/background/providers/impl/aggregating_aurora_factors/ErrorHandlingTask.java
package se.gustavkarlsson.aurora_notifier.android.background.providers.impl.aggregating_aurora_factors;

import java.util.concurrent.Callable;
import java.util.concurrent.TimeoutException;

interface ErrorHandlingTask<V> {
	Callable<V> getCallable();
	V handleInterruptedException(InterruptedException e);
	V handleThrowable(Throwable e);
	V handleTimeoutException(TimeoutException e);
}

File: app/src/main/java/se/gustavkarlsson/skylight/android/background/providers/impl/aggregating_aurora_factors/ErrorHandlingTask.kt
package se.gustavkarlsson.skylight.android.background.providers.impl.aggregating_aurora_factors

import java.util.concurrent.Callable
import java.util.concurrent.TimeoutException

internal interface ErrorHandlingTask<V> {
    val callable: Callable<V>
    fun handleInterruptedException(e: InterruptedException): V
    fun handleThrowable(e: Throwable): V
    fun handleTimeoutException(e: TimeoutException): V
}
